src/Distortion.cpp

```diff
@@ -185,13 +186,18 @@ void removeDistortion(double dx, double dy, double &ux, double &uy,
       //           b0 +b1*r +b2*r^2 +b3*r^3
       //   v[Z] = INS<INSTID>BORESIGHT[Z] .
 
-      // Coeffs should be [x0,x1,x2,x3,y0,y1,y2,y3]
-      if (opticalDistCoeffs.size() != 8) {
-        throw "Distortion coefficients for Kaguya TC must be of size 8, got: " +  std::to_string(opticalDistCoeffs.size());
+      // Coeffs should be [boresightX,x0,x1,x2,x3,boresightY,y0,y1,y2,y3]
+      if (opticalDistCoeffs.size() != 10) {
+        csm::Error::ErrorType errorType = csm::Error::INDEX_OUT_OF_RANGE;
+        std::string message = "Distortion coefficients for Kaguya LISM must be of size 10, got: " +  std::to_string(opticalDistCoeffs.size());
+        std::string function = "removeDistortion";
+        throw csm::Error(errorType, message, function);
       }
 
-      const double* odkx = opticalDistCoeffs.data();
-      const double* odky = opticalDistCoeffs.data()+4;
+      double boresightX = opticalDistCoeffs[0];
+      std::vector<double> odkx(opticalDistCoeffs.begin()+1, opticalDistCoeffs.begin()+5);
+      double boresightY = opticalDistCoeffs[5];
+      std::vector<double> odky(opticalDistCoeffs.begin()+6, opticalDistCoeffs.begin()+10);
```

Comment:
Switched to use std vectors and range constructors for better error checking. We've already had problems with reading off the end of C arrays. So, I figure it's safer to move to std vectors here because the input data is already a std vector.